src/reader/linearring.rs

```diff
@@ -40,17 +41,49 @@ pub struct LinearRing<'a> {
 
 impl<'a> LinearRing<'a> {
     pub fn new(buf: &'a [u8], byte_order: Endianness, offset: u64, dim: Dimension) -> Self {
+        Self::try_new(buf, byte_order, offset, dim).unwrap()
+    }
+
+    pub fn try_new(
```

Comment:
Could we make this `pub(crate)`?

Reply:
Unless there is a reason to only expose a panic-ing version to users, I'm inclined to leave this `pub` method.

Reply:
None of these types are intended for direct use by end users. And indeed these types aren't even exported. 
<img width="659" alt="image" src="https://github.com/user-attachments/assets/34eb4c32-b34b-428b-854a-1152497ed21a" />

but the structs must be `pub` because they're used in the public implementation of `GeometryTrait` by `Wkb`

Reply:
So actually the `new` function should be `pub(crate)` as well (I think all the other `new` constructors are `pub(crate)` already). I don't think that would be a breaking change since `Linearring` isn't currently exported. (You could technically access it via geo traits associate types somehow)

Reply:
Created #75 so we can just merge this PR